webpack.config.js

```diff
@@ -0,0 +1,160 @@
+const path = require("path");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const MonacoWebpackPlugin = require("monaco-editor-webpack-plugin");
+const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
+const { ProvidePlugin } = require("webpack");
+
+const OUTPUT_FOLDER = path.join(__dirname, "build");
+const CODEBASE = path.join(__dirname, "source");
+const isProd = process.env.NODE_ENV === "prod";
+
+const railsTransformer = (mode) => ({
+  loader: "shell-loader",
+  options: {
+    script: `bundle exec ruby transformer.rb ${mode}`,
+    cwd: "./rails",
+    maxBuffer: Math.pow(1024, 3)
+  }
+});
+
+const htmlExporter = {
+  loader: "file-loader",
+  options: {
+    name: "[path][name].html",
+  },
+};
+
+const assetExporter = (regex, folder) => ({
+  test: regex,
+  use: [{
+    loader: "file-loader",
+    options: {
+      outputPath: folder,
+      name: "[name].[ext]",
+      publicPath: `/${folder}`,
+    }
+  }],
+});
+
+module.exports = {
+  context: CODEBASE,
+
+  devServer: {
+    contentBase: OUTPUT_FOLDER,
+    compress: true,
+    port: 4567
+  },
+
+  entry: {
+    vendor: "./javascripts/vendor.js",
+    main: "./javascripts/index.js",
+  },
+
+  optimization: {
+    minimizer: isProd ? [
+      new UglifyJsPlugin({
+        uglifyOptions: {
+          cache: true,
+          parallel: true,
+          mangle: false
```

Comment:
we can't mangle our variable names, since we omitted them in angular DI :( it uses function argument names now for lookup so it's impossible to minify entirely. Might worth writing some regex replace as a next step